flexget/plugins/input/anidb_feed.py

```diff
@@ -0,0 +1,479 @@
+            raise plugin.PluginError('Unable to reach anidb feed url: %s' % url)
+
+        fixed_xml = self.make_feedparser_friendly(r.content)
+        try:
+            if fixed_xml:
+                parsed_xml = feedparser.parse(fixed_xml)
+            else:
+                parsed_xml = feedparser.parse(r.content)
+            xml_feed = parsed_xml.feed
+            if 'error' in xml_feed:
+                if 'code' in xml_feed['error']:
+                    if 'description' in xml_feed['error']:
+                        log.error(
+                            'Error code: %s detail: %s' % (xml_feed['error']['code'], xml_feed['error']['description']))
+                    else:
+                        log.error('Error code: %s' % xml_feed['error']['code'])
+        except Exception as ex:
+            log.error('Unable to parse the XML from url: %s error: %s' % (url, ex))
+            return []
+
+        if not len(parsed_xml.entries) > 0:
+            log.info('No entries returned from xml.')
+            return []
+
+        entries = self.parse_from_xml(parsed_xml.entries, config)
+        if len(entries) == 0:
+            log.verbose('No entries parsed from xml.')
+        # else:
+        #   dump_entry(entries[0])
+
+        return entries
+
+    def make_feedparser_friendly(self, data):
+        try:
+            dom = minidom.parseString(data)
+            entries = dom.getElementsByTagName('entry')
+            for entry in entries:
+                items_ns = entry.getElementsByTagNameNS(NAMESPACE_URL, NAMESPACE_TAGNAME)
+                if items_ns:
+                    for node_ns in items_ns:
+                        dt_nodes = node_ns.getElementsByTagName(NAMESPACE_NAME + ':dt')
+                        dd_nodes = node_ns.getElementsByTagName(NAMESPACE_NAME + ':dd')
+                        if len(dt_nodes) != len(dd_nodes):
+                            logging.warning('unexpected nodecount')
+                            continue
+                        for idx, dt_node in enumerate(dt_nodes):
+                            if len(dt_node.childNodes) >= 1 and len(dd_nodes[idx].childNodes) >= 1:
+                                if dt_node.childNodes[0].nodeValue is not None and dd_nodes[idx].childNodes[0].nodeValue is not None:
+                                    dt_node.tagName = NAMESPACE_NAME + ':%s' % dt_node.childNodes[0].nodeValue.replace(' ', '_')
+                                    dt_node.setAttribute('name', dt_node.childNodes[0].nodeValue)
+                                    dt_node.setAttribute('value', dd_nodes[idx].childNodes[0].nodeValue)
+                                    dt_node.removeChild(dt_node.childNodes[0])
+                                    node_ns.removeChild(dd_nodes[idx])
+                                    entry.appendChild(dt_node)
+        except Exception as ex:
+            logging.debug('Unable to rename nodes in XML: %s' % ex)
+            return None
```

Comment:
> Way too wide exception catch that completely hides crashes.

That's actually the point in this case, this purely optional fixup code, so if anything is not as expected we just skip and use the unfixed version of the xml.

> You are returning empty list or None in many error cases. This is bad design, for error cases one should rely exceptions, not return values.

Thats because i cant see/find a way to enforce a contract with the caller, to handle my exceptions. What i see is a consensus that code checks vs None or work with empty lists.

> Check if python has some good builtin exception to be used, or if this is fatal raise PluginError

That seems reasonable, but my problem is that outside of a top level error, why should i raise a exception for a unimportant field/operation of a single entry, if i could instead just skip this field/entry and continue working on the rest of the entries?